plugins/ssh/ssh.plugin.zsh

```diff
@@ -0,0 +1,39 @@
+############################################################
+# Take all host sections in .ssh/config and offer them for
+# completion as hosts (e.g. for ssh, rsync, scp and the like)
+# Filter out wildcard host sections.
+local hosts
+if [[ -f $HOME/.ssh/config ]]; then
+  hosts=($(egrep '^Host.*' $HOME/.ssh/config | awk '{print $2}' | grep -v '^*' | sed -e 's/\.*\*$//'))
+  zstyle ':completion:*:hosts' hosts $hosts
+fi
+
+############################################################
+# Remove host key from known hosts based on a host section
+# name from .ssh/config
+function ssh_rmhkey {
+  if [[ "x$1" == "x" ]]; then return; fi
+  ssh-keygen -R $(grep -A10 $1 ~/.ssh/config | grep -i HostName | head -n 1 | awk '{print $2}')
+}
+compctl -k hosts ssh_rmhkey
+
+############################################################
+# Load SSH key into agent
+function ssh_load_key() {
+  local key=$1
+  if [[ "$key" == "" ]]; then return; fi
```

Comment:
Here we could use `[[ -z "$key" ]]` it's maybe more standard, but it's also ok to leave this like that.

Reply:
Personally I prefer the `-z` form and consider it more standard by far. If using this form, I'd expect to see it in old-school Bourne style, as

```bash
if [[ x"$key" == x"" ]]
```

...to avoid testing any genuinely empty strings.